linkerd/app/core/src/serve.rs

```diff
@@ -39,39 +39,38 @@ pub async fn serve<M, S, I, A>(
-                                    Ok(mut accept) => {
-                                        match accept
-                                            .call(io::ScopedIo::server(io))
-                                            .err_into::<Error>()
-                                            .await
-                                        {
-                                            Ok(()) => debug!("Connection closed"),
-                                            Err(reason) if is_io(&*reason) => {
-                                                debug!(%reason, "Connection closed")
-                                            }
-                                            Err(error) => info!(%error, "Connection closed"),
+                    // Dispatch all of the work for a given connection onto a
+                    // connection-specific task.
+                    tokio::spawn(
+                        async move {
+                            match accept.ready_oneshot().err_into::<Error>().await {
+                                Ok(mut accept) => {
+                                    match accept
+                                        .call(io::ScopedIo::server(io))
+                                        .err_into::<Error>()
+                                        .await
+                                    {
+                                        Ok(()) => debug!("Connection closed"),
+                                        Err(reason) if is_io(&*reason) => {
+                                            debug!(%reason, "Connection closed")
                                         }
-                                        // Hold the service until the connection is complete. This
-                                        // helps tie any inner cache lifetimes to the services they
-                                        // return.
-                                        drop(accept);
-                                    }
-                                    Err(error) => {
-                                        warn!(%error, "Server failed to become ready");
+                                        Err(error) => info!(%error, "Connection closed"),
                                     }
+                                    // Hold the service until the connection is complete. This
+                                    // helps tie any inner cache lifetimes to the services they
+                                    // return.
+                                    drop(accept);
+                                }
+                                Err(error) => {
+                                    warn!(%error, "Server failed to become ready");
                                 }
                             }
-                            .in_current_span(),
-                        );
-                    });
+                        }
+                        .instrument(span.exit().or_current()),
```

Comment:
And this exits the above-entered span? and then we use that or the parent?

Reply:
yup!